modules/data.land/inst/doanload_NEON_multi_site.R

```diff
@@ -0,0 +1,111 @@
+#' Title generalized for download NEON files of multiSites. and automatically detect the most recent available NEON data and if data is problematic, try next year.
+#'
+#' @param settings a multisite settings object, should work with a single site setting. 
+#' @param main_store_dir the preferred path to store the downloaded NEON files. the log file and VEG_INFO should also be stored in that path
+#'
+#' @return VEG_INFO: a list object containing veg_info, site IDs of each site.
+#' @export log.txt file describing the progress of download
+
+Download_multiSites_NEON <- function(settings, main_store_dir){
+  #get NEON sites table
+  neonsites <- neonstore::neon_sites(api = "https://data.neonscience.org/api/v0", .token = Sys.getenv("NEON_TOKEN"))
+  
+  #output
+  VEG_INFO <- list()
+  
+  #log file
+  log_txt <- c()
+  
+  #starting FOR LOOP
+  for (i in 1:length(settings)) {
+    #match lon and lat
+    tempLat <- as.numeric(settings[[i]]$run$site$lat)
+    tempLon <- as.numeric(settings[[i]]$run$site$lon)
+    
+    betyneondist <- swfscMisc::distance(lat1 = tempLat, lon1 = tempLon, lat2 = neonsites$siteLatitude, lon2 = neonsites$siteLongitude)
+    index <- which(betyneondist == min(betyneondist))
+    tempNEON <- neonsites[index,]
+    
+    #grab data info from NEON
+    dataProducts <- tempNEON$dataProducts[[1]]
+    
+    #create subfolder
+    store_dir <- paste0(main_store_dir,"/",settings[[i]]$run$site$id)
+    dir.create(store_dir)
+    
+    #filter on date
+    if(length(which(dataProducts$dataProductCode=="DP1.10098.001")) == 0){
+      print(paste0("no data available for site: ", tempNEON$siteCode, ". jump to the next site!"))
+      log_txt <- c(log_txt, paste0("no data available for site: ", tempNEON$siteCode, ". jump to the next site!"))
```

Comment:
1. In this approach to creating a log file, the log only gets written if there are no errors anywhere in the loop. Better to write logs incrementally
2. Also better to use the PEcAn.logger package for writing logs

Reply:
fixed, but not using PEcAn.logger

Reply:
please update to use PEcAn.logger